### DISCUSSION TITLE
Confused about using CLI args when using runnables/debuggables

### DISCUSSION THREAD

**Discussion by Overloader6** [mrcjkb/rustaceanvim#237] [Q&A] [answered]
So how do I run my application with cli args? When typing `rustLsp runnables arg1 arg2` I get `runnables: unexpected arguments: { "arg1", "arg2" }`

**Comment by mrcjkb** [comment]
Which version of the plugin are you using?
The ability to pass args was introduced in version 4.

**Reply by Overloader6** [reply]
I installed it with [nixvim](https://github.com/nix-community/nixvim/tree/main), not sure about the version. is there some way I can query it?

**Reply by mrcjkb** [reply, marked as answer]
The current version on nixos-unstable is 3.9.6, so that's probably what you're using with nixvim.
If you want to use a newer version, you can either PR an update to `luaPackages.rustaceanvim` in nixpkgs, or import this repo's flake and set `rustaceanvim.package`.

**Reply by Overloader6** [reply]
I see, thanks! I went with the flake part (although im still a noob at flakes. is there a better way to get the output than `inputs.rustaceanvim.packages.x86_64-linux.default`) and now running works. still need to set up a deug adapter but thats a different problem

**Reply by mrcjkb** [reply]
Using the package is fine 😀
An alternative option would be to apply this flake's overlay to your nixpkgs instance, but that's just a matter of personal preference.

For debugging, I would recommend codelldb, which you can install via the `vscode-extension-vadimcn-vscode-lldb.adapter` package (the `.adapter` package is codelldb without the vscode extension).
